### ISSUE TITLE
Incorrect formatting output for header when using frontmatter

### ISSUE THREAD

**Issue by scscgit** [remarkjs/remark-lint#253]
### Subject of the issue

When formatting a following `a.md` file
```md
---
a
c
---
```

via `npx remark-cli a.md --output` (or just reading without `--output`) (or `npx remark a.md` with remark installed), its content becomes
```md
***

## a&#xD;&#xA;c
```

### Your environment

*   **OS**: Windows 10, tested also in WSL: Ubuntu
*   **Packages**: `npx remark --version` says `remark: 13.0.0, remark-cli: 9.0.0`

*   **Env**: `npm --version` says `6.14.8`

### Steps to reproduce

https://codesandbox.io/s/fast-flower-xmrb5?file=/a.md

Run the command via terminal
```
npx remark-cli a.md
npx: installed 127 in 50.802s
***

## a&#xD;&#xA;c
a.md: no issues found
```

### Expected behavior

Anything but this

### Actual behavior

`&#xD;&#xA;` and a questionable header translation

Notes:
* this doesn't occur when testing via JS in https://codesandbox.io/s/remark-debug-ikwvx
* this issue occurs by default to users of https://github.com/nuxt/create-nuxt-app with content module in `content/hello.md` file
* is there a chance you could add a test to cover this via GitHub Actions?
* the new output would fail MD041 of markdownlint, which isn't reflected in [doc/comparison-to-markdownlint.md](doc/comparison-to-markdownlint.md)
  * and the initial ## seems to pass heading-increment

**Comment by ChristianMurphy** [comment]
@scscgit to clarify, you want

```md
---
a
c
---
```

to be interpreted as frontmatter? or as a header?
In [commonmark](https://commonmark.org/) (the specification remark follows) it would be interpreted as a header, which is the behavior you are seeing.
If you want it to be interpreted as frontmatter, you need to add https://github.com/remarkjs/remark-frontmatter

For comparison:
without remark-frontmatter: https://astexplorer.net/#/gist/cf75cb128acef741e05e5a6ec5e52675/deccb2b86156435a51cd7aa64e8fb91d358413f5
with remark-frontmatter: https://astexplorer.net/#/gist/cf75cb128acef741e05e5a6ec5e52675/120d8c9e21cc370511b95fd7bb419cb8d2dc7bc3

**Comment by ChristianMurphy** [comment]
> I wasn't aware of this incompatibility

It is still compatible, it just needs to be enabled through a plugin.

> Does this mean the `&#xD;&#xA;` should be added as a correct behavior when using commonmark rather than keeping the content as it is?

yes, remark is an [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) rather than a [CST](https://en.wikipedia.org/wiki/Parse_tree).
Meaning the structure of the content will be preserved, but the formatting is not guaranteed to be the same.

`&#xD;&#xA;` is an encoded softbreak, which matches with the commonmark reference AST https://spec.commonmark.org/dingus/?text=---%0Aa%0Ac%0A---%0A

> I haven't noticed any mention about such setup in remark-lint README

`remark-frontmatter` is a remark plugin, it can be enabled the same way as other plugins.
How you  run the linter changes how you enable plugins.
Assuming you are using remark cli, https://github.com/remarkjs/remark-lint#configuring-remark-lint shows how to enable plugins
